tutorial/README.md

```diff
@@ -0,0 +1,824 @@
+
+Now paste the IP address into your browser (your IP address will be different from the one shown here). After a few seconds you should see some basic info about the server.
+
+If this is working then you've successfully completed this step. You can now remove this application and move on the next step.
+
+```sh
+# using the control plane VM
+azoam hello-world/* | kubectl delete -f -
+```
+
+## Deploying Flight-Tracker
+
+Now you're ready to work with the flight tracker application, and write some of your own OAM definitions.
+
+The application being deployed is shown in the following diagram:
+
+![Application architecture diagram](service-tracker-diagram.jpg)
+
+---
+
+For this example we'll be running the PostgreSQL database in the cluster as a `ContainerizedWorkload`. This means that you'll need a total of 6 `Component` definitions and a single `ApplicationConfiguration`.
+
+You can find some starter manifests in `./flight-tracker` - the `quakes-api` service and database are done for you to introduce more of the concepts. You can divide all of these assets into individual files however you like. The tools don't enforce any restrictions. It's most convenient to place all of the files in a single directory so they can be applied using a directory wildcard.
+
+You can also find the completed sample in `./flight-tracker-solution` for reference if you want to check your answers.
+
+### 1. Create an empty ApplicationConfiguration
+
+Create an `ApplicationConfiguration` declaration with the name `flight-tracker`. Remember that an `ApplicationConfiguration` should use the resource type `ApplicationConfiguration@core.oam.dev/v1alpha2`. Feel free to refer to the previous example `./hello-world/applicationconfiguration.arm` as a template.
+
+For now the `spec.components` node should be an empty array.
+
+### 2. Using the db Component
+
+The database in this example is run as a normal `ContainerizedWorkload`. You can find the complete code in `./flight-tracker/db.arm`. This step will require you to understand the new features used by this component, and include it in your `ApplicationConfiguration`.
+
+Here are the contents of the `db` `Component` with some annotations to introduce the parameter concept. Most of this definition should look familiar from the examples with the hello-world application.
+
+```js
+// input statements declare parameters that can be set by the ApplicationConfiguration
+//
+// Use parameters to avoid hardcoding of environment-specific configuration. In this
+// the database username, database name, and database password will be passed in. Since
+// we're running the database in a container, these values will be used to configure
+// the database itself. Components that need to connect to the database will need the
+// same values to make a connection.
+//
+// Most parameters will be of type `string`. Use `securestring` when the contents are
+// sensitive.
+input string dbuser
+input string dbname
+input securestring dbpassword
```

Comment:
Have you tried this with azoam? I am not sure if this type works or not.

Reply:
It works. I don't think it does anything different from string.